js/src/data/actions.js

```diff
@@ -1247,3 +1248,35 @@ export function* upsertTour( tour, upsertingClientStoreFirst = false ) {
 		);
 	}
 }
+
+export function* fetchGoogleAdsAccountStatus() {
+	try {
+		const { data } = yield fetchWithHeaders( {
+			path: `${ API_NAMESPACE }/ads/account-status`,
+		} );
+
+		return {
+			type: TYPES.RECEIVE_GOOGLE_ADS_ACCOUNT_STATUS,
+			data,
+		};
+	} catch ( response ) {
+		const errorBodyPromise = response?.json() || response?.text();
+		const errorData = yield awaitPromise( errorBodyPromise );
+
+		if ( response.status === 428 ) {
+			// Update to the data store as the "errorData" is valid account status data.
+			return {
+				type: TYPES.RECEIVE_GOOGLE_ADS_ACCOUNT_STATUS,
+				data: errorData,
+			};
+		}
```

Comment:
It no longer needs to use `fetchWithHeaders` and detect 428 status code after https://github.com/woocommerce/google-listings-and-ads/pull/2270/commits/bb66dd4a43d67eb3b213fdcff86f1955a08706df.

Reply:
Handled in 5d2a502.